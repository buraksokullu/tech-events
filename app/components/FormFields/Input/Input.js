import React from 'react';
import PropTypes from 'prop-types';

import s from './Input.scss';

const Input = ({ input, type, ...args }) => (
  <input {...input} {...args} type={type} className={s.input} />
);

Input.propTypes = {
  input: PropTypes.node,
  type: PropTypes.string
};

Input.defaultProps = {
  input: null,
  type: 'text'
};

export default Input;
