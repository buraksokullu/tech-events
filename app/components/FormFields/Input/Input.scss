.input {
  background-color: $grey30;
  border: 2px solid $grey30;
  border-radius: 8px;
  box-sizing: border-box;
  color: $grey110;
  font-size: 16px;
  font-style: normal;
  font-weight: normal;
  letter-spacing: -0.32px;
  line-height: 1.5;
  outline: 0;
  padding: 10px 14px;
  width: 100%;
  box-sizing: border-box;

  &:focus {
    background-color: $white;
    border-color: $orange;
    color: $grey110;
    transition: all 250ms ease;
  }
}
