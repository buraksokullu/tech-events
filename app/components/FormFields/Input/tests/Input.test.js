import React from 'react';
import { shallow } from 'enzyme';
import Input from '../Input';

describe('Input', () => {
  let inputProps = null;

  beforeEach(() => {
    inputProps = {
      input: 'input',
      type: 'type'
    };
  });

  it('Should render correctly', () => {
    const component = shallow(<Input {...inputProps} />);

    expect(component).toMatchSnapshot();
  });
});
