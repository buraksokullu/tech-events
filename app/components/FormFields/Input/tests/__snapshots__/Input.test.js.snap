// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should render correctly 1`] = `
<input
  0="i"
  1="n"
  2="p"
  3="u"
  4="t"
  type="type"
/>
`;
